sentry/src/main/java/io/sentry/GsonSerializer.java

```diff
@@ -136,37 +136,14 @@ public GsonSerializer(
     }
   }
 
-  /**
-   * Serialize a SentryEvent to a stream Writer (JSON)
-   *
-   * @param event the SentryEvent
-   * @param writer the Writer
-   * @throws IOException an IOException
-   */
   @Override
-  public void serialize(final @NotNull SentryEvent event, final @NotNull Writer writer)
+  public <T> void serialize(final @NotNull T event, final @NotNull Writer writer)
       throws IOException {
     Objects.requireNonNull(event, "The SentryEvent object is required.");
     Objects.requireNonNull(writer, "The Writer object is required.");
-
-    gson.toJson(event, SentryEvent.class, writer);
-    writer.flush();
-  }
-
-  /**
-   * Serialize a Session to a stream Writer (JSON)
-   *
-   * @param session the Session
-   * @param writer the Writer
-   * @throws IOException an IOException
-   */
-  @Override
-  public void serialize(final @NotNull Session session, final @NotNull Writer writer)
-      throws IOException {
-    Objects.requireNonNull(session, "The Session object is required.");
-    Objects.requireNonNull(writer, "The Writer object is required.");
-
-    gson.toJson(session, Session.class, writer);
+    // TODO: check if debug level is enabled
+    logger.log(SentryLevel.DEBUG, "Serializing object: %s", gson.toJson(event));
```

Comment:
Here ideally we log the serialized object but we need to check if debug logs are enabled as this comes with a performance cost.